src/biogeophys/SnowCoverFractionMod.F90

```diff
@@ -0,0 +1,785 @@
+module SnowCoverFractionMod
+
+  !---------------------------------------------------------------------------
+  ! !DESCRIPTION:
+  ! Classes for various methods of computing snow cover fraction
+  !
+  ! !USES:
+#include "shr_assert.h"
+  use shr_kind_mod   , only : r8 => shr_kind_r8
+  use shr_log_mod    , only : errMsg => shr_log_errMsg
+  use shr_infnan_mod , only : nan => shr_infnan_nan, assignment(=)
+  use shr_nl_mod     , only : shr_nl_find_group_name
+  use shr_mpi_mod    , only : shr_mpi_bcast
+  use fileutils      , only : getavu, relavu, opnfil
+  use abortutils     , only : endrun
+  use decompMod      , only : bounds_type
+  use spmdMod        , only : masterproc, mpicom
+  use clm_varcon     , only : rpi
+  use clm_varctl     , only : iulog, use_subgrid_fluxes
+  use ncdio_pio      , only : file_desc_t
+  use paramUtilMod   , only : readNcdioScalar
+  use ColumnType     , only : column_type
+  use glcBehaviorMod , only : glc_behavior_type
+  use landunit_varcon, only : istice_mec
+  implicit none
+  save
+  private
+  !
+  ! !PUBLIC TYPES:
+  public :: CreateAndInitScfMethod
+  public :: snow_cover_fraction_type
+
+  type, abstract :: snow_cover_fraction_type
+     private
+   contains
+     ! Initialize this instance
+     procedure(Init_Interface), deferred :: Init
+
+     ! Update snow depth and snow fraction
+     procedure(UpdateSnowDepthAndFrac_Interface), deferred :: UpdateSnowDepthAndFrac
+
+     ! Add new snow to integrated snow fall
+     procedure(AddNewsnowToIntsnow_Interface), deferred :: AddNewsnowToIntsnow
+
+     ! Single-point function: return fractional snow cover during melt
+     procedure(FracSnowDuringMelt_Interface), deferred :: FracSnowDuringMelt
+  end type snow_cover_fraction_type
+
+  abstract interface
+
+     subroutine Init_Interface(this, bounds, col, glc_behavior, NLFilename, params_ncid)
+       use decompMod, only : bounds_type
+       use ColumnType, only : column_type
+       use glcBehaviorMod, only : glc_behavior_type
+       use ncdio_pio, only : file_desc_t
+       import :: snow_cover_fraction_type
+
+       class(snow_cover_fraction_type), intent(inout) :: this
+       type(bounds_type), intent(in) :: bounds
+       type(column_type), intent(in) :: col
+       type(glc_behavior_type), intent(in) :: glc_behavior
+       character(len=*), intent(in) :: NLFilename ! Namelist filename
+       type(file_desc_t), intent(inout) :: params_ncid ! pio netCDF file id for parameter file
+     end subroutine Init_Interface
+
+     subroutine UpdateSnowDepthAndFrac_Interface(this, bounds, num_c, filter_c, &
```

Comment:
Personally I still like to have at least a short description of the subroutine

Reply:
Done in 3daf3066